File: p2.py
#Time: O(N)
#Space: O(1)
class Solution:
    def jump(self, nums: List[int]) -> int:
        if len(nums)==0  or len(nums)==1:
            return 0
        cur_int = nums[0]
        next_int = nums[0]
        jumps = 1
        n = len(nums)
        for i in range(1,n):
            next_int = max(next_int,nums[i]+i)
            if i < n-1 and i == cur_int:
                cur_int = next_int
                jumps+=1
        
        return jumps
                
                
                
            
        
        

File: p2.java
// Time Complexity : O(n)
// Space Complexity :O(1)
// Did this code successfully run on Leetcode : yes
// Any problem you faced while coding this :no


// Your code here along with comments explaining your approach

class Solution {
    public int jump(int[] nums) {
        if(nums.length < 2) return 0;
        int currInt = nums[0];
        int nextInt = nums[0];
        int jumps = 1;
        for(int i=1; i<nums.length; i++){
            nextInt = Math.max(nextInt, i+nums[i]);

            if(i != nums.length-1 && i==currInt){
                currInt = nextInt;
                jumps++;
            }
        }

        return jumps;
    }
}